examples/IMDb/leap_binder.py

```diff
@@ -0,0 +1,257 @@
+from typing import List, Dict
+import pandas as pd
+import numpy as np
+import numpy.typing as npt
+import tensorflow as tf
+
+# Tensorleap imports
+from code_loader import leap_binder
+from code_loader.contract.datasetclasses import PreprocessResponse
+from code_loader.contract.enums import LeapDataType
+from code_loader.contract.visualizer_classes import LeapText, LeapHorizontalBar
+
+from IMDb.config import CONFIG
+from IMDb.data.preprocess import download_load_assets
+from IMDb.gcs_utils import _download
+from IMDb.utils import prepare_input, prepare_input_dense_model
+
+
+# Preprocess Function
+def preprocess_func() -> List[PreprocessResponse]:
+    """
+    Performs data preprocessing and prepares training and validation data for a machine learning model.
+
+    :return: A list of PreprocessResponse objects containing preprocessed data and associated information.
+    """
+    tokenizer, df = download_load_assets()
+    train_label_size = int(0.9 * CONFIG['NUMBER_OF_SAMPLES'] / 2)
+    val_label_size = int(0.1 * CONFIG['NUMBER_OF_SAMPLES'] / 2)
+    df = df[df['subset'] == 'train']
+    train_df = pd.concat([df[df['gt'] == 'pos'][:train_label_size], df[df['gt'] == 'neg'][:train_label_size]],
+                         ignore_index=True)
+    val_df = pd.concat([df[df['gt'] == 'pos'][train_label_size:train_label_size + val_label_size],
+                        df[df['gt'] == 'neg'][train_label_size:train_label_size + val_label_size]], ignore_index=True)
+    ohe = {"pos": [0., 1.0], "neg": [1.0, 0.]}
+
+    train = PreprocessResponse(length=2 * train_label_size, data={"df": train_df, "tokenizer": tokenizer, "ohe": ohe})
```

Comment:
Maybe limit the train_label_size and val_label_size by len(df['gt'] == LABEL)? If someone places a number of samples which is greater than df size we might have a bug here since length would be incorrect

Reply:
change to len(df)